lib/evaluation/evaluationClient.ts

```diff
@@ -83,11 +86,11 @@ export class EvaluationClient {
     uriEvaluation
       .setNext(new InputDescriptorFilterEvaluationHandler(this))
       .setNext(new PredicateRelatedFieldEvaluationHandler(this))
-      .setNext(new MarkForSubmissionEvaluationHandler(this))
       .setNext(new LimitDisclosureEvaluationHandler(this))
       .setNext(new SubjectIsIssuerEvaluationHandler(this))
       .setNext(new SubjectIsHolderEvaluationHandler(this))
-      .setNext(new SameSubjectEvaluationHandler(this));
+      .setNext(new SameSubjectEvaluationHandler(this))
+      .setNext(new MarkForSubmissionEvaluationHandler(this));
```

Comment:
The reordering was done because the previous implementation ignored all the evaluation handlers that come after mark for submission (step 3). So the results of limit disclosure, same subject, subject is holder and subject is issuer were never considered during the creation of the presentation submission.

Reply:
The results of the following evaluators should not have been ignored. But also my comment stands. I would have preferred the same implementations as it was mentioned in the specs.
Nonetheless, the current implementation is acceptable.